apps/spruce/src/pages/task/logs/utils.ts

```diff
@@ -114,6 +114,37 @@ export const validateParams = (
   };
 };
 
+export interface ValidatedTestLogParams {
+  taskId: string;
+  execution: number;
+  testName: string;
+}
+
+export const validateTestLogParams = (
+  taskId: string | undefined,
+  execution: string | null,
+  testName: string | null,
+): ValidatedTestLogParams => {
+  if (!taskId) {
+    throw new Error("Task ID not specified");
+  }
+
+  const executionNum = parseInt(execution || "", 10);
+  if (isNaN(executionNum)) {
+    throw new Error("Execution not specified");
+  }
```

Comment:
This behavior differs from the page its replacing. The legacy viewer will auto default to the latest execution if execution is not defined
https://github.com/evergreen-ci/evergreen/blob/a7c38531aa6c3b40793371d70728e73bb5f8a9bf/rest/route/task_output.go#L41

Reply:
I'm not sure what you mean, removing `2` from this URL or replacing it with a higher value doesn't work https://evergreen.mongodb.com/test_log/evergreen_ui_spruce_e2e_spruce_0_patch_8f0d16fffcdc5a5d5b5327fb4078de46d2a946c6_691c9e1fba439a0007bbc16c_25_11_18_16_26_09/2?test_name=3b047941ae47e3937c7f677bd147810d#L0